### ISSUE TITLE
Graphical corruption on launcher

### ISSUE THREAD

**Issue by K4sum1** [olauncher/olauncher#12]
I sometimes get graphical corruption on the launcher, I'm not sure why or where I could find any logs or anything that I could add to help. All I can think of is that my GPU is a 2080 Ti, but I'm sure that if this happened on 2000 series cards, more people would have it.
![ScreenShot_20221010112147](https://user-images.githubusercontent.com/77039240/194911877-c8bcb5dd-ab96-4574-8c4d-e249f5477f74.png)
![ScreenShot_20221010112140](https://user-images.githubusercontent.com/77039240/194911923-7783914b-4650-4141-8a31-c6f8a622b2ca.png)

**Comment by bigfoot547** [comment]
Hmm. Is this because the controls are placed incorrectly or the window painting just glitches out? What causes these corruptions, and do they fix themselves when you drag the window off the screen then back on?

**Comment by K4sum1** [comment]
> Hmm. Is this because the controls are placed incorrectly or the window painting just glitches out? What causes these corruptions, and do they fix themselves when you drag the window off the screen then back on?

It appears to happen when I open the profile dropdown. The controls are still in the proper places, and I can use them, but moving my mouse around makes the glitches change.

Here's a video.
https://user-images.githubusercontent.com/77039240/195143661-cb4a037b-cb0d-4d3a-80ba-d9a91f1117f9.mp4

**Comment by bigfoot547** [comment]
It may have something to do with the JVM you're using, because I can't reproduce it with openJDK or adoptium. Maybe try one of those?

**Comment by K4sum1** [comment]
> It may have something to do with the JVM you're using, because I can't reproduce it with openJDK or adoptium. Maybe try one of those?

I thought that might have been the case, I was originally using Liberica 17, but I swapped to Adoptium and still got it before making the issue.

**Comment by K4sum1** [comment]
Hmm, I've temporarily swapped my card with an AMD one to do some testing, and there's no more issues with the launcher.

**Comment by bigfoot547** [comment]
I'm glad it works for you! Please reopen if the issues reappear.

**Comment by K4sum1** [comment]
Uhh what I meant is that my AMD card doesn't give me the issues that my Nvidia card did. So there might still be a problem, but I won't be going back to my Nvidia card for a day or two. Also I can't reopen issues closed by the author because GitHub moment.

**Comment by bigfoot547** [comment]
Fair enough. But I can't help but think that the issue is possibly upstream from this project, since changing your videocard fixed it.

**Comment by K4sum1** [comment]
So I went back to my Nvidia card, and it has the same issue, on a fresh OS install.

**Comment by bigfoot547** [comment]
I am still of the mind that this is an issue upstream of my project. Have you tried other applications that use java swing? Do they work properly?

**Comment by K4sum1** [comment]
Minecraft itself is fine, the only other thing I can think of would be the forge installer, but I don't think I tested it with the card. I won't be able to test anything until tomorrow at minimum.

**Comment by K4sum1** [comment]
I went back to the AMD card to do a little more testing, and now it does it too apparently.

**Comment by bigfoot547** [comment]
Please reopen this issue if you can test other swing applications.

**Comment by K4sum1** [comment]
What is a swing application?

**Comment by bigfoot547** [comment]
A java application that uses swing

**Comment by bigfoot547** [comment]
Turns out it is related to java's use of direct3d on some graphics cards. See [this SO answer](https://stackoverflow.com/a/872018) for a workaround. I will try to make a hotfix out of this soon. Thanks for reporting, I'm glad this is finally hopefully resolved.

**Comment by exrodev** [comment]
Hi @K4sum1 does this issue still exist for you? I pushed something that may fix it but I'm unable to test it since I cannot recreate this issue.

**Comment by K4sum1** [comment]
@exrodev It seems to be weird in that it comes and goes. I'm still running 1.7.2_09 and haven't bothered to update yet. It does seem to be gone even in this old version. Will need to see if I can get my laptop or other Windows install to replicate it, then try updating.

**Comment by exrodev** [comment]
We haven't created a new release with the possible fix yet but if you do manage to replicate this, let me know and I could send you a version that has the fix on Discord.
